src/templates/globals/util/breadcrumbUtils.tsx

```diff
@@ -0,0 +1,66 @@
+import { BreadcrumbItem } from '@/types/dataTypes/drupal/field_type'
+
+export function deriveLastBreadcrumbFromPath(
```

Comment:
i would move this file & its tests to `@lib/utils/` instead of `@templates/globals/util` because they aren't actually returning templates, only modifying data

Reply:
I think I raised a similar issue on the previous PR, though I can't find it now. I think there's an argument to having utility modules that are specific to a thing residing with that thing. I recently implemented this with `src/drupal/lovell/utils.ts` as a collection of utility functions that are specific to Lovell and that do not fall under the other files (i.e. `constants.ts`, `staticPaths.ts`, etc.). 

But I also think this situation is slightly different. I think there's a strong argument for having the functions currently in this file reside in `@/lib/utils/breadcrumbs.ts` because they are not necessarily specific to this template. Breadcrumb utils feel more general than a single template. If there _were_ a utility function that were somehow only for a single template (something like `formatBreadcrumbForMyVerySpecialUseCase`) then we could perhaps have a `utils` directory within that template directory, but, honestly, in that case I'd probably keep that function in the component file.

In any case, I think we should settle on `utils` rather than `util` since that is what is used elsewhere already.

Reply:
I would really like them placed in `@lib/utils` solely because I think there's a future where we do a lot of the business-logic-breadcrumb processing inside of the queries files, instead of at the breadcrumb component level.

I'd love that future breadcrumb component to be as dumb as possible, just taking a list of breadcrumbs and displaying them. It should be up to the query to determine what format those breadcrumbs should be delivered in (hidden or custom home crumb, whatever)